native/src/main/java/io/ballerina/stdlib/crypto/PgpEncryptionGenerator.java

```diff
@@ -0,0 +1,145 @@
+import org.bouncycastle.openpgp.PGPPublicKeyRing;
+import org.bouncycastle.openpgp.PGPPublicKeyRingCollection;
+import org.bouncycastle.openpgp.PGPUtil;
+import org.bouncycastle.openpgp.operator.jcajce.JcaKeyFingerprintCalculator;
+import org.bouncycastle.openpgp.operator.jcajce.JcePGPDataEncryptorBuilder;
+import org.bouncycastle.openpgp.operator.jcajce.JcePublicKeyKeyEncryptionMethodGenerator;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.security.SecureRandom;
+import java.security.Security;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.Optional;
+
+public class PgpEncryptionGenerator {
+
+    static {
+        if (Objects.isNull(Security.getProvider(BouncyCastleProvider.PROVIDER_NAME))) {
+            Security.addProvider(new BouncyCastleProvider());
+        }
+    }
+
+    private final int compressionAlgorithm;
+    private final int symmetricKeyAlgorithm;
+    private final boolean armor;
+    private final boolean withIntegrityCheck;
+    private static final int BUFFER_SIZE = 8192;
+
+    public PgpEncryptionGenerator(int compressionAlgorithm, int symmetricKeyAlgorithm, boolean armor,
+                             boolean withIntegrityCheck) {
+        this.compressionAlgorithm = compressionAlgorithm;
+        this.symmetricKeyAlgorithm = symmetricKeyAlgorithm;
+        this.armor = armor;
+        this.withIntegrityCheck = withIntegrityCheck;
+    }
+
+    public void encryptStream(OutputStream encryptOut, InputStream clearIn, long length, InputStream publicKeyIn)
+            throws IOException, PGPException {
+        PGPCompressedDataGenerator compressedDataGenerator =
+                new PGPCompressedDataGenerator(compressionAlgorithm);
+        PGPEncryptedDataGenerator pgpEncryptedDataGenerator = new PGPEncryptedDataGenerator(
+                // Configure the encrypted data generator
+                new JcePGPDataEncryptorBuilder(symmetricKeyAlgorithm)
+                        .setWithIntegrityPacket(withIntegrityCheck)
+                        .setSecureRandom(new SecureRandom())
+                        .setProvider(BouncyCastleProvider.PROVIDER_NAME)
+        );
+        // Add public key
+        pgpEncryptedDataGenerator.addMethod(new JcePublicKeyKeyEncryptionMethodGenerator(
+                getPublicKey(publicKeyIn)));
+        if (armor) {
+            encryptOut = new ArmoredOutputStream(encryptOut);
+        }
+        OutputStream cipherOutStream = pgpEncryptedDataGenerator.open(encryptOut, new byte[BUFFER_SIZE]);
+        copyAsLiteralData(compressedDataGenerator.open(cipherOutStream), clearIn, length);
+        compressedDataGenerator.close();
+        cipherOutStream.close();
+        encryptOut.close();
```

Comment:
Rather than manually doing this it is better to use `try-with-resources` block to close the created IO streams.